libs/wingc/src/lsp/ast_traversal.rs

```diff
@@ -0,0 +1,887 @@
+use lsp_types::Position;
```

Comment:
If you're looking at this file, you may be thinking: "Hey shouldn't there be some `impl Iterator`s in here?"

You are correct, but iterating over our tree is pretty complicated due to how non-uniform and recursive it is.

I tried it and couldn't finish the task, so I did this instead. I'm open to help on it if I'm just being dumb and it's not that crazy.

Reply:
Definitely something I'd be down to try

Reply:
@Chriscbr Sweet, as a followup to this PR or as part of this one?

Reply:
Either is cool with me 😄